mousestyles/density_estimation.py

```diff
@@ -0,0 +1,162 @@
+# coding: utf-8
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from mousestyles.data import load_movement
+import numpy as np
+import pandas as pd
+
+
+def extract_distances(strain, mouse, day, step=1e2):
+    """
+    Return a numpy array object of project movement data
+    for the specified combination of strain, mouse and day.
+
+    At regular timesteps, defined by the step parameter,
+    compute the euclidian distance between the positions
+    of the mouse at two consecutive times.
+
+    More specifically:
+
+    - let delta_t be the step parameter.
+    - let t_n be the sequance of non negative numbers such
+    that t_0 = 0 and t_(n+1) = t_n + delta_t. The sequence
+    is defined for all n such that n>=0 and t_n <= time
+    of the experiment
+    - let d_n be the sequence of non negative numbers such
+    that d_0 = 0 and d_n equals the position of the mouse
+    at a particular day at time t_n. d_n is then defined
+    on the same set of integers as the sequence t_n.
+    - The function returns the sequence d_n.
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: int
+        nonnegative integer indicating the mouse number
+    day: int
+        nonnegative integer indicating the day number
+    step: float
+        positive float defining the time between two observations
+
+    Returns
+    -------
+    movement : numpy array
+
+    Examples
+    --------
+    >>> dist = extract_distances(0, 0, 0, step=1e2)
+    0
+    """
+    movement = load_movement(strain, mouse, day)
+    # Compute distance between samples
+    dist = np.empty((movement.shape[0]-1, 2))
+    x = np.array(movement["x"])
+    y = np.array(movement["y"])
+    # dist[:, 0] contains the distances between two
+    # consecutive points
+    dist[:, 0] = np.sqrt((x[1:] - x[:-1])**2 + (y[1:] - y[:-1])**2)
```

Comment:
This could probably be shorter 
dist = np.sqrt(movement["x"].diff() ** 2 + movement["y"].diff ** 2)